### DISCUSSION TITLE
Precision Mod

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#28708] [Ideas]
A mod that ties your actual aim to your accuracy. As exactly as it says.
Like all rhythm games have accuracy judgements, This mod introduces adds aiming to that judgement. Combining both on tapping and on aim. _As if osu standard isn't focusing on aim already_

Here's an illustration. 

![Untitled](https://github.com/ppy/osu/assets/64581009/974ab4ef-0161-461f-a307-d24006edd3ce)

It's simple as you get. Don't tap on green or orange (100/50) and all you need to do is to aim dead center on the circles and get more accuracy and score. Simple as that.

Except it's kinda not. **There's a Scoring concern.** _And possibly the mod giving Performance Points (and other mods) is still undecided:_

- Should it be Difficulty Increase mod or just Fun mod? (Decision between having score multiplier or not)
- Should the mod split the accuracy as ratio between aiming and tapping? Or independent entirely?
- Should it even be independent accuracy? and only give more score in addition to tapping accuracy's score?
- Should the mod even have a customization?
- Is the mod's aim accuracy judgement supposed to scale with the beatmap's OD?

These are some only concerns that I have. In my opinion, I believe we can just leave the mod as Fun and tinker with it whatever as it is because, well... It's fun! 
I hope this gets someone's interest and make it real

**Comment by shiumano** [comment]
It looks like McOsu's Target Mod.

I've been doing some light reading of McOsu's source code,

Deviation from center is less than 50% and timing is GREAT or OK: GREAT
Deviation from center is less than 70%: OK
Deviation from center is less than 95%: MEH
If all conditions are not met: MISS

The judgment seems to be made in this way.

This mod probably belongs in the conversion or fun category; if you use McOsu's method, it's hard to see how it would be an increase in difficulty, since you may get a GREAT score with less accuracy.

Of course, I do not believe this is the absolute right answer. It is just one of the answers that we have.
We could ignore McOsu and make our own.

**Reply by NotGumballer91** [reply]
We should make our own. That's why lazer is built from the ground up in the first place.

> since you may get a GREAT score with less accuracy.

I dont understand what could this mean. Is it assuming that hitting a circle at 300 would give 300 regardless if you tapped at 100 or 50?

**Reply by shiumano** [reply]
I apologize for the lack of clarity in my explanation.
It hits the center of the hit circle and the timing is 300s or 100s.
If it was 50s, it will be 100s if you hit the center.

Well, I think there are many more fun ways to judge than this, so you may not have to go into it too much.

**Reply by NotGumballer91** [reply]
I'm more leaning towards the separate accuracy thing. 
Even if you reach hit all 300s (100% accuracy) as tapping
You would would have overall 75% accuracy if aimed all 100s

That's 50/50. Or the weight can be different like 75/25

**Comment by Tenexxt** [comment]
It would most likely be a mod in the "fun" category,. because it's kinda too hard for diff increase imo. On easier maps when you have a lot of time to react, it would be a fun challenge, but on like 6* and up... I have some doubts. Of course the mod could still be added, I'm just explaining why I don't think it would be added as a "main" difficulty increase mod.
About score multiplier, some fun mods do also have it, so if it was to be one, it's still fine. I would give this mod a .06 multiplier, because the HR and HD can also be stupidly hard depending on the map, so it's forgivable, probably not enough to be a "main" mod, but one that could give pp.

**Comment by ashandelle** [comment]
This mod should be added catch too

I think it should be a difficulty increasing mod but only if it considers both normal accuracy and precision, if it only considers precision then it should be a fun mod. I think it should give pp especially if it becomes a difficulty increasing mod.

**Reply by Tenexxt** [reply]
I've actually suggested an accuracy heatmap for osu!catch (#29920) and since it was said it could be added later down the line, with it such mod could come. Tho since catch has only 2 judgements (GREAT and MISS), unless the mod would completely alter the ruleset judgement settings, it would be impossible for the judgements to be something else, so you could only alter in the mods customisation settings the accuracy you'd have to catch the fruit with for it to register as a great and not a miss.

Now, this is still a *fun* idea IMO, but not what really fits under the same thing as this proposal.

**Comment by ashandelle** [comment]
I think they should have separate accuracies, although I'm not sure how to display the overall accuracy to the player. If they were completely separated it would require having another accuracy skin element. Maybe by default the existing accuracy element will display 2 percentages accuracy on top and precision on bottom for new players who don't know how to skin, but give the player the ability to separate them. As for showing the judgment, right now the animation has the judgment fall downward, an additional judgment could be added for precision that falls upward.

**Reply by ashandelle** [reply]
In catch precision can likely just replace accuracy since there are no judgments and it just shows percentage hit

**Reply by ashandelle** [reply]
For leaderboard scoring if you want to include precision you have to combine it with accuracy. There are 2 main ways I'm thinking of, you can average them arithmetically (a+b/2) or geometrically (sqrt(a*b)). Geometrically punishes an imbalance more than arithmetically and you can weight it using powers instead of factors. I think that punishing imbalance would be better for a ranked mod, but if its just a fun mod then it doesn't matter how their combined and arithmetic is much simpler for players to understand.

**Reply by ashandelle** [reply]
I didn't realize the judgment only falls on misses but I think the idea is still good

**Comment by ashandelle** [comment]
An implementation of this should probably take #34350 and the new hit display into consideration
